game-core/src/main/java/games/strategy/triplea/image/ImageTransformer.java

```diff
@@ -0,0 +1,103 @@
+package games.strategy.triplea.image;
+
+import java.awt.Color;
+import java.awt.geom.AffineTransform;
+import java.awt.image.AffineTransformOp;
+import java.awt.image.BufferedImage;
+
+/**
+ * Class used to transform images including colorizing and flipping. Colorizing is based on finding
+ * the luminance from the original image then applying hue, saturation, and brightness to colorize
+ * it. More information about luminance can be found here:
+ * https://en.wikipedia.org/wiki/Relative_luminance.
+ */
+public class ImageTransformer {
+
+  private static final int MAX_COLOR = 256;
+  private static final float LUMINANCE_RED = 0.2126f;
+  private static final float LUMINANCE_GREEN = 0.7152f;
+  private static final float LUMINANCE_BLUE = 0.0722f;
+
+  private ImageTransformer() {}
+
+  /**
+   * Apply color and brightness to the given image. This uses the hue and saturation of the given
+   * color. It ignores the brightness of the color and uses the given brightness value.
+   */
+  public static void colorize(final Color color, final int brightness, final BufferedImage image) {
+    final float[] hsb = Color.RGBtoHSB(color.getRed(), color.getGreen(), color.getBlue(), null);
+    colorize((int) (hsb[0] * 360), (int) (hsb[1] * 100), brightness, image);
+  }
+
+  /**
+   * Apply HSB values to the given image.
+   *
+   * @param hue 0 to 360
```

Comment:
Does the game disallow the out of range values anywhere? Beyond the javadoc, would a precondition check here help ensure we do not continue with invalid input?

Reply:
Everything but brightness is derived from Color (which is already validated) so those values have to be valid. I added a validation when reading the brightness property to ensure its within the allowed range. I think those checks are sufficient for now. If this class would become a library or be used in other areas then adding validation of parameters would probably be a good idea.

Reply:
Given it's a public API, the defensive programming part of me, and the 'fail early' principle makes me feel like the parameters should be validated, given they are not constrained by type. There is a calculation that occurs to get us to a HSB value, any bugs there could send incorrect data here. Having the checks would nip that in the bud. Any such incorrect values perhaps would not go on to be detected and we'd get bad rendering without necessarily knowing why. I'd rather see the bug revealed in testing as an exception then later post-deployment.

Reply:
My mistake, it is a private API. I'm not used to seeing fully formal javadocs on private methods.

Looking at that now, I wonder a bit why the two methods are not combined now. Going a bit further, it is a bit odd that for example we multiply 'hue' by 360 to then just divide it by 360.

Reply:
The problem is really that the java color apis deal with decimal representation of HSB values which isn't very common. You could argue to combine them now though I think there is value in keeping them separate as the actual algorithm is meant to work on HSB values and for TripleA it just makes more sense to use color code and brightness. Once I did that the original HSB method is no longer called outside the class so is private but I think still worth keeping as is.

Reply:
Ok. :+1: 
 IMO this is an opportunity to make the 2 line method, where one line is another method call, take on more responsibility. At that point 'colorize' would go to a package protected module so that it's usage is then encapsulated.